ci: Remove deprecated container.greedy

maflcko opened the pull request:
The option is to be phased out, so remove it to avoid relying on it. Update container.cpu where needed.

hebasto commented:
> The option is to be phased out,

It is still mentioned in https://cirrus-ci.org/guide/linux/#linux-containers.

Commit:
ci: Remove deprecated container.greedy

The option is to be phased out, so remove it to avoid relying on it.
Update container.cpu and timeouts where needed.

fanquake reviewed (approved):
ACK fa248a4c616be31d231c671e8feb0dbb46ac54cd

hebasto commented:
> For some reason the jammy macOS taks is uncacheable? [cirrus-ci.com/task/4744875699077120?logs=ci#L2486](https://cirrus-ci.com/task/4744875699077120?logs=ci#L2486)

See:
- https://github.com/bitcoin/bitcoin/issues/21552
- https://github.com/bitcoin/bitcoin/pull/24620

hebasto reviewed (approved):
ACK fac14c4e498f2d38b8b337d4c145129d07403a6d.

Mentioning "timeouts" in the commit message looks unrelated.
